text/0000-multiarch-builders-and-package.md

```diff
@@ -0,0 +1,1119 @@
+it will be removed
+- When `targets` is not present in [builder.toml](https://buildpacks.io/docs/reference/config/builder-config/), throw a warning messages indicating
+  a new `--target` flag is available
+- Keep doing our current process to create a builder
+
+### To improve user experience
+
+We propose:
+
+- Add `targets` section to the `builder.toml` schema, this will keep consistency for end-users to understand how to 
+define multi-architecture. Adding more than one target to the `builder.toml` will be considered by `pack` as an 
+acknowledgement of the desired to generate multi-arch [Builders](https://buildpacks.io/docs/concepts/components/builder/).  
+
+The new schema will be
+similar to:
+```toml
+# Buildpacks to include in builder, 
+# MUST point to an Image Index that matches targets
+[[buildpacks]]
+uri = "<some uri>"
+
+[run]
+# Runtime images 
+# MUST point to an Image Index that matches targets
+[[run.images]]
+image = "<run image reference>"
+
+[build]
+# This image is used at build-time
+# MUST point to an Image Index that matches targets
+image = "<build image reference>"
+
+# Target platforms to support with the Builder
+[[targets]]
+os = "<operating system>"
+arch = "<system architecture>"
+variant = "<architecture variant>"
+[[targets.distributions]]
+name = "<distribution ID>"
+versions = ["<distribution version>"]
+```
+- Add a new `--target` optional flag with format `[os][/arch][/variant]:[name@version]` to create a builder for a 
+particular target, this will help end-users to specify the platform for which they want to create single OCI artifact.
+
+### Examples
+
+Let's use some examples to explain the expected behavior in different use cases
+
+#### `Targets` are not present in `builder.toml`
+
+This is probably the case for most of the Buildpack Authors, for example 
+[Paketo](https://github.com/paketo-buildpacks/builder-jammy-tiny/blob/main/builder.toml), lets suppose a`buildpack.toml` 
+like:
+
+```toml
+# Buildpacks to include in builder
+[[buildpacks]]
+uri = "<some buildpacks>"
+
+# Order used for detection
+[[order]]
+[[order.group]]
+id = "<some order>"
+version = "<some version>"
+
+[stack]
+id = "io.buildpacks.samples.stacks.jammy"
+build-image = "cnbs/sample-base-build:jammy"
+run-image = "cnbs/sample-base-run:jammy"
+```
+Or we include `build` and `run` images
+
+```toml
+# Base images used to create the builder
+[build]
+image = "cnbs/sample-base-build:jammy"
+[run]
+[[run.images]]
+image = "cnbs/sample-base-run:jammy"
+```
+
+In these cases, the expected output will be similar to:
+
+```bash
+pack builder create <builder> --config ./builder.toml 
+Warning: "stack" has been deprecated, prefer "targets" instead: https://github.com/buildpacks/rfcs/blob/main/text/0096-remove-stacks-mixins.md
+Warning: A new '--target' flag is available to set the target platform for the builder, using 'linux/amd64' as default
```

Comment:
This assumes that the build image is a manifest list, no? Because if the build image is linux/arm64 (only) then we create an arm64 image, no?

Reply:
Yeah, the `pack builder create` command assumes the build, run and buildpack images are multi-arch, and when building for a particular platform, `pack` should validate it